django/core/management/commands/showmigrations.py

```diff
@@ -84,9 +86,15 @@ def show_list(self, connection, app_names=None):
                         title = plan_node[1]
                         if graph.nodes[plan_node].replaces:
                             title += " (%s squashed migrations)" % len(graph.nodes[plan_node].replaces)
+                        # Get the applied migration instance
+                        filtered = filter(lambda m: m == plan_node, loader.applied_migrations)
```

Comment:
This was a problem before as well but I guess it should be fixed while you are around.

In the case where `showmigrations` is not called with an app label (or called with multiple ones) this will incur `len(app_names)` queries since each access to `loader.applied_migrations` incurs one. It might be worth assigning a `applied_migrations = loader.applied_migrations` variable outside of the loop.

Reply:
Can do.

Reply:
Wait, `loader.applied_migrations` is not the same as `Recorder.applied_migrations()` where the latter actually makes a DB query. `loader.applied_migrations` is a property that doesn't make any DB queries, but gets set in `build_graph()`

Reply:
Ahh never mind then, I assumed it was the same function based on the name.